ovs/extensions/storageserver/storagedriver.py

```diff
@@ -284,49 +284,63 @@ def load(self):
-                        reloaded = True
-                    except Exception as ex:
-                        if 'ClusterNotReachableException' not in str(ex):
-                            raise
-                else:
-                    self._logger.info('Applying storagedriver configuration changes on {0}'.format(client.ip))
-                    try:
-                        with remote(client.ip, [LocalStorageRouterClient]) as rem:
-                            changes = copy.deepcopy(rem.LocalStorageRouterClient(self.remote_path).update_configuration(self.remote_path))
-                            reloaded = True
-                    except Exception as ex:
-                        if 'ClusterNotReachableException' not in str(ex):
-                            raise
-                if reloaded is True:
-                    if isinstance(changes, dict):
-                        for change in changes:
-                            if change['param_name'] not in self.dirty_entries:
-                                raise RuntimeError('Unexpected configuration change: {0}'.format(change['param_name']))
-                            self._logger.info('Changed {0} from "{1}" to "{2}"'.format(change['param_name'], change['old_value'], change['new_value']))
-                            self.dirty_entries.remove(change['param_name'])
-                        self._logger.info('Changes applied')
-                    if len(self.dirty_entries) > 0:
-                        self._logger.warning('Following changes were not applied: {0}'.format(', '.join(self.dirty_entries)))
-                else:
-                    self._logger.warning('Changes were not applied since storagedriver is unavailable')
+
+        # No changes detected in the configuration management
+        if len(self.dirty_entries) == 0 and force_reload is False:
+            self._logger.debug('No need to apply changes, nothing changed')
+            self.is_new = False
+            return changes
+
+        # Retrieve the changes from volumedriver
+        self._logger.info('Applying local storagedriver configuration changes{0}'.format('' if client is None else ' on {0}'.format(client.ip)))
+        try:
+            if client is None:
+                changes = LocalStorageRouterClient(self.remote_path).update_configuration(self.remote_path)
             else:
-                self._logger.debug('No need to apply changes, nothing changed')
+                with remote(client.ip, [LocalStorageRouterClient]) as rem:
+                    changes = copy.deepcopy(rem.LocalStorageRouterClient(self.remote_path).update_configuration(self.remote_path))
+        except Exception as ex:
+            if 'ClusterNotReachableException' not in str(ex):
+                raise
+
+        # No changes
+        if len(changes) == 0:
+            self._logger.warning('Changes were not applied since storagedriver is unavailable')
```

Comment:
This message might also pop up if there were dirty entries, but they could not be applied by the SD